[rachelyeohm] iP

rachelyeohm opened the pull request:
# Nyabot
> Nya nya nya nya nya. 🐱 

Have you ever wanted a perfectly normal task manager, except it says _nya_? Here you go!

Features:

- [x] Add task
- [x] Delete task
- [x] List tasks
- [x] Mark task done or undone
- [x] Finding tasks
- [ ] Loading and saving chatbot data
- [ ] Saying nya

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```
And lastly...
- **nya**
- *nya*
- nya
1. nya
2. nya
3. nya
4. [nya](https://youtu.be/dQw4w9WgXcQ?si=05jBZ4vwvHZgYYqa)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

rachelyeohm committed:
Add hello and bye functions, remove mentions of Duke

rachelyeohm committed:
Add functionality for echoing input + abstract horizontal lines in string

rachelyeohm committed:
Implement storing and displaying texts in list form

rachelyeohm committed:
Add Task class and ability to mark done/undone

rachelyeohm committed:
Add different types of task

rachelyeohm committed:
Add semi-automated testing with .bat file

rachelyeohm committed:
Add error handling with exceptions

rachelyeohm committed:
Add delete functionality

rachelyeohm committed:
Add enumeration for mark status

rachelyeohm committed:
Add load and save functions

rachelyeohm committed:
Merge branch 'branch-Level-7'

rachelyeohm committed:
Add classes for more OOP structure

rachelyeohm committed:
Implement OOP structure

rachelyeohm committed:
Implement more OOP structure

rachelyeohm committed:
Merge branch 'A-MoreOOP'

rachelyeohm committed:
Add date parsing and output formatting

rachelyeohm committed:
Merge branch 'branch-Level-8'

rachelyeohm committed:
Add packages

rachelyeohm committed:
Merge branch 'A-Packages'

rachelyeohm committed:
Merge remote-tracking branch 'remotes/origin/add-gradle-support'

rachelyeohm committed:
Add JUnit tests for Save, ToDo and Parser class

rachelyeohm committed:
Merge branch 'A-JUnit'

rachelyeohm committed:
Edit build.gradle file to match package structure and create jar file

rachelyeohm committed:
Merge branch 'A-Jar'

rachelyeohm committed:
Add find command

rachelyeohm committed:
Add Javadoc header comments for classes and methods

rachelyeohm committed:
Merge branch 'A-JavaDoc'

rachelyeohm committed:
Add Javadoc headers for methods and classes of Command and its subclasses

rachelyeohm committed:
Merge branch 'A-JavaDoc'

rachelyeohm committed:
Adhere to coding standard

rachelyeohm committed:
Merge branch 'branch-A-CodingStandard'

rachelyeohm committed:
Update README.md

rachelyeohm committed:
Update README.md

rachelyeohm committed:
Updae README.md

rachelyeohm committed:
Add GUI functionality

rachelyeohm committed:
Merge branch 'branch-Level-10'

AgentHagu commented on src/main/java/command/Command.java:
```diff
@@ -0,0 +1,33 @@
+ */
+abstract public class Command {
+
+    protected boolean isExit = false;
```
Good naming convention used for boolean with the "is" prefix

AgentHagu commented on src/main/java/command/FindCommand.java:
```diff
@@ -0,0 +1,40 @@
+     */
+    @Override
+    public String execute(TaskList taskList, Ui ui, Storage storage) {
+        TaskList taskList1 = new TaskList();
```
You could consider renaming `taskList1` to something more descriptive, it might get confused with `taskList`.

AgentHagu commented on src/main/java/components/Parser.java:
```diff
@@ -0,0 +1,229 @@
+        String firstWord = input.split(" ")[0];
+        switch(firstWord) {
+        case "bye":
+            return new ExitCommand();
```
I like how you didn't indent the `case` statements in your switch-case, following the coding standards. Good job

AgentHagu commented on src/main/java/components/Parser.java:
```diff
@@ -0,0 +1,229 @@
+     * @throws NyabotMissingArgumentException If no content is written after the command
+     * in the input.
+     */
+    public static String parseTaskWithWord(String input, String command) throws NyabotMissingArgumentException {
```
Should this method be public? I noticed that it was only used within the Parser class itself.

AgentHagu commented on src/main/java/task/Task.java:
```diff
@@ -0,0 +1,107 @@
+    }
+
+    private boolean isDone;
+    private String taskName;
```
I like how you kept the class variables private, following the coding standard. Good job!

AgentHagu commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,71 @@
+import java.util.ArrayList;
+
+
+public class TaskList {
```
Perhaps you could add JavaDoc header comments for this class file? I noticed that most of the other classes have header comments but this class's was missing

AgentHagu commented on src/main/java/task/TaskList.java:
```diff
@@ -0,0 +1,71 @@
+        return textListString.toString();
+    }
+
+    public String displayTasksSaveable() throws NyabotParseException {
```
Perhaps you could rename this method? Currently, it doesn't seem that the method displays the strings, but instead returns them, so "display" may not be a suitable term for this method.

AgentHagu commented on src/test/java/components/ParserTest.java:
```diff
@@ -0,0 +1,158 @@
+public class ParserTest {
+
+    @Test
+    public void parseNewTask_todo_expectedBehavior() throws NyabotException {
```
I like how the test method names follow the three part format in the coding standard. Good job!

AgentHagu reviewed (commented):
Overall, good job following the coding standard! 

I like how a good majority of the class files had JavaDoc header comments to explain the class and the public methods. There are only a few nitpicks to fix.

althea28 commented on src/main/java/command/Command.java:
```diff
@@ -0,0 +1,33 @@
+     *
+     * @return Boolean whether the command is an exit command.
+     */
+    public boolean isExit() {
```
Perhaps can consider renaming this method to a verb for clarity or to avoid confusion with boolean variable isExit?

althea28 commented on src/main/java/components/Parser.java:
```diff
@@ -0,0 +1,229 @@
+        case "list":
+            return new ListCommand();
+        case "mark": case"unmark":
+            int number = handleTaskWithTaskNumber(input, firstWord);
```
Perhaps can consider specifying this variable name to include what this number represents?

althea28 commented on src/main/java/components/Parser.java:
```diff
@@ -0,0 +1,229 @@
+            if (parts.length < 3 || parts[1].trim().isEmpty() || parts[2].trim().isEmpty()) {
+                throw new NyabotMissingArgumentException("A valid /from and /to command is required, nya!");
+            }
+            String[] partsFull = input.split(" ");
```
Perhaps can consider renaming this collection of Strings to be of plural form?

althea28 commented on src/main/java/components/Storage.java:
```diff
@@ -0,0 +1,117 @@
+            String line = br.readLine();
+
+            while (line != null) {
+                String[] taskArray = line.split(" \\| ");
```
Perhaps can consider renaming this array of Strings to be of plural form?

althea28 reviewed (commented):
Almost all code quality guidelines were included, with only a few very minor outstanding cases. Also I like the name of your chatbot!! 😺

rachelyeohm committed:
Add assertions statements

The code lacks assertions, making error detection during development less effective and increasing the risk of unnoticed bugs.

Errors need to be caught early to ensure code robustness and maintainability.

Let's add assertions to validate critical assumptions within the code. Use assertions to provide immediate feedback when conditions are violated.

This method is chosen for its simplicity and effectiveness in catching errors during development without impacting production performance.

Assertions will improve code reliability and aid in quicker debugging.

rachelyeohm committed:
Merge pull request #1 from rachelyeohm/branch-A-Assertions

Add assertions statements

rachelyeohm committed:
Add Scheduler Feature to Show Events on Date

The to-do list app lacks a scheduling feature that allows users to view
events filtered by specific dates. Users cannot easily see what tasks or
 events are scheduled for a particular day, which reduces the app’s
 effectiveness in managing time-sensitive tasks.

A scheduler is needed to provide a clear view of events, helping users
manage their daily schedules more effectively and stay organized.

Let's implement a scheduler command that accepts a specific date
(schedule YYYY-MM-DD) to display events starting and ending on that day.
 Additionally, allow the command schedule today to show events scheduled
  for the current day.

This approach offers flexibility and enhances usability by making it
easy to filter and view relevant events based on the user's query,
improving the overall user experience.

The new feature will help users stay on top of their tasks and events,
making the app more valuable for daily planning.

rachelyeohm committed:
Merge pull request #2 from rachelyeohm/branch-B-ViewSchedules

Add Scheduler Feature to Show Events on Date

rachelyeohm committed:
Fix code of poor quality

rachelyeohm committed:
Merge pull request #3 from rachelyeohm/branch-A-CodeQuality

Fix code of poor quality

rachelyeohm committed:
Add tags

rachelyeohm committed:
Update README

rachelyeohm committed:
Update README

rachelyeohm committed:
Update README

rachelyeohm committed:
Update README

rachelyeohm committed:
Update README

rachelyeohm committed:
Merge pull request #5 from rachelyeohm/FixREADME

Fix readme

rachelyeohm committed:
Update README.md

The README.md lacks detailed documentation of all available functions
and commands.

Without this information, new contributors and users may find it
difficult to understand or utilize the project's full capabilities.

Let's update the README.md to include a comprehensive list of all
functions and commands, along with brief descriptions for each.

This approach provides clarity and eases the onboarding process by
offering a single, accessible reference for key functionality.

README.md updates are crucial for improving project maintainability
and user experience.

rachelyeohm committed:
Merge pull request #6 from rachelyeohm/FixREADME

Update README.md

rachelyeohm committed:
Fix typo

rachelyeohm committed:
Merge pull request #7 from rachelyeohm/FixREADME

Fix typo

rachelyeohm committed:
Refactor main Nyabot class

rachelyeohm committed:
Add doc Ui.png

rachelyeohm committed:
Update docs README.md

rachelyeohm committed:
Update README.md

rachelyeohm committed:
Update README.md

rachelyeohm committed:
Update README.md

rachelyeohm committed:
Update README.md

rachelyeohm committed:
Update README.md

rachelyeohm committed:
Update README.md

rachelyeohm committed:
Merge pull request #8 from rachelyeohm/Release

Update README.md and release

rachelyeohm committed:
Update README.md

rachelyeohm committed:
Merge pull request #9 from rachelyeohm/rachelyeohm-patch-1

Update README.md

rachelyeohm committed:
Update README.md

rachelyeohm committed:
Update README.md

rachelyeohm committed:
Update README.md

rachelyeohm committed:
Update README.md
